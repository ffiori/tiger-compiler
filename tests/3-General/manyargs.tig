let
 function printint (i : int) =
   let function f (i : int) =
        if i > 0
	then (f (i/10); print (chr (i-i/10*10+ord ("0"))))
   in if i < 0 then (print ("-"); f (-i))
      else if i>0 then f (i)
      else print ("0")
   end

 function printmany (a1: int, a2: int, a3: int, a4: int,
   a5: int, a6: int, a7: int, st0: int, st1: int,
   st2: int, st3: int, st4: int) : int = (if st4 <> 0 then (
    print("a1: "); printint(a1); print("\n");
    print("a2: "); printint(a2); print("\n");
    print("a3: "); printint(a3); print("\n");
    print("a4: "); printint(a4); print("\n");
    print("a5: "); printint(a5); print("\n");
    print("a6: "); printint(a6); print("\n");
    print("a7: "); printint(a7); print("\n");
    print("st0: "); printint(st0); print("\n");
    print("st1: "); printint(st1); print("\n");
    print("st2: "); printint(st2); print("\n");
    print("st3: "); printint(st3); print("\n");
    print("st4: "); printint(st4); print("\n"); /* for x:= 0 to 10000000 do (x;()); */
    printmany(a1, a2, a3, a4, a5, a6, a7, st0, st1, st2, st3, st4-1))
   else 0)

in printmany(1,2,3,4,5,6,7,100,101,102,103,104)
end
