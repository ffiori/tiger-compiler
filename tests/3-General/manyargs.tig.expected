a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 104
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 103
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 102
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 101
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 100
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 99
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 98
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 97
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 96
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 95
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 94
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 93
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 92
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 91
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 90
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 89
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 88
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 87
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 86
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 85
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 84
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 83
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 82
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 81
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 80
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 79
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 78
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 77
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 76
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 75
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 74
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 73
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 72
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 71
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 70
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 69
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 68
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 67
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 66
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 65
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 64
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 63
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 62
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 61
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 60
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 59
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 58
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 57
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 56
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 55
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 54
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 53
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 52
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 51
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 50
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 49
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 48
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 47
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 46
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 45
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 44
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 43
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 42
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 41
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 40
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 39
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 38
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 37
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 36
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 35
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 34
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 33
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 32
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 31
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 30
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 29
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 28
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 27
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 26
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 25
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 24
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 23
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 22
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 21
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 20
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 19
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 18
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 17
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 16
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 15
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 14
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 13
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 12
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 11
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 10
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 9
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 8
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 7
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 6
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 5
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 4
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 3
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 2
a1: 1
a2: 2
a3: 3
a4: 4
a5: 5
a6: 6
a7: 7
st0: 100
st1: 101
st2: 102
st3: 103
st4: 1
